fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/audit/BulkAuditLogger.java

```diff
@@ -0,0 +1,312 @@
+import com.ibm.fhir.audit.AuditLogService;
+import com.ibm.fhir.audit.AuditLogServiceFactory;
+import com.ibm.fhir.audit.beans.ApiParameters;
+import com.ibm.fhir.audit.beans.AuditLogEntry;
+import com.ibm.fhir.audit.beans.Batch;
+import com.ibm.fhir.audit.beans.Context;
+import com.ibm.fhir.audit.beans.Data;
+import com.ibm.fhir.config.FHIRConfigHelper;
+import com.ibm.fhir.config.FHIRConfiguration;
+import com.ibm.fhir.config.FHIRRequestContext;
+import com.ibm.fhir.core.FHIRUtilities;
+import com.ibm.fhir.core.util.handler.IPHandler;
+import com.ibm.fhir.model.resource.Bundle;
+import com.ibm.fhir.model.resource.Resource;
+import com.ibm.fhir.model.util.FHIRUtil;
+
+/**
+ * This class provides convenience methods for FHIR Java Batch Jobs
+ * that access and manipulate FHIR Data, thus needing to write FHIR audit log entries.
+ */
+public class BulkAuditLogger {
+
+    /*
+     * Identifies the component when writing to the IBM FHIR Server
+     */
+    private static final String COMPONENT_ID = "fhir-bulkdata-server";
+
+    private static final String CLASSNAME = BulkAuditLogger.class.getName();
+    private static final Logger log = java.util.logging.Logger.getLogger(CLASSNAME);
+
+    private static final IPHandler componentIpHandler = new IPHandler();
+
+    private AuditLogService svc = AuditLogServiceFactory.getService();
+
+    public BulkAuditLogger() {
+        // No Operation
+    }
+
+    public boolean shouldLog() {
+        // Wraps common code for logging
+        if (log.isLoggable(Level.FINE)) {
+            log.fine("Bulk Data Audit Log is off");
+        }
+        return svc.isEnabled();
+    }
+
+    /*
+     * This method is common code in the audit logging.
+     */
+    private void log(AuditLogEventType eventType, String action, String description, Resource oldResource, Resource newResource, Date startTime, Date endTime,
+        Response.Status responseStatus, String queryString, Long totalSearch, String location, String users) throws Exception {
+
+        AuditLogEntry entry = createAuditLogEntry(eventType, newResource, startTime, endTime, responseStatus, location, users);
+        entry.getContext().setAction(action);
+        entry.setDescription(description);
+
+        // Only add if it's a search
+        if (AuditLogEventType.FHIR_SEARCH.equals(eventType)) {
+            entry.getContext().setQueryParameters(queryString);
+            entry.getContext().setBatch(Batch.builder().resourcesRead(totalSearch).build());
+        }
+
+        svc.logEntry(entry);
+    }
+
+    /*
+     * Creates an audit log entry with attributes common to all the bulkdata jobs.
+     * @param eventType - the type of event
+     * @param resource
+     * - The Resource object.
+     * @param startTime
+     * - The start time of the request execution.
+     * @param endTime
+     * - The end time of the request execution.
+     * @param responseStatus
+     * - The response status.
+     * @param location
```

Comment:
Applies to several methods in this file:  Location and users params do not have hyphen in comments like other params.

Reply:
Fixed, coming in PR